Expose QUnit version as property and display it in HTML Reporter

shivamdixit opened the pull request:
Screenshot:

![qunit](https://cloud.githubusercontent.com/assets/3061095/6544759/bd9018f0-c584-11e4-8488-815d068da95b.png)

Ref #554

jzaefferer commented:
Showing the QUnit version in a separate line takes up too much visual space. Maybe put it in front of the user-agent, separated by a semicolon?

shivamdixit commented:
@jzaefferer The length becomes too large if we display it in same line. Is it OK?

jzaefferer commented on src/core.js:
```diff
@@ -184,6 +185,10 @@ config.modules.push( config.currentModule );
+
+	// Expose the current QUnit version
+	// `version` initialized at top of scope
+	QUnit.version = version;
```
Since `version` isn't used anywhere else, it can be inlined, that is, use `"@VERSION"` here.

jzaefferer commented on reporter/html.js:
```diff
@@ -454,12 +455,23 @@ function storeFixture() {
 
+function appendVersion() {
+	var version = id( "qunit-version" ),
+		value = "@VERSION";
```
Can this use `QUnit.version` instead?

jzaefferer commented:
> The length becomes too large if we display it in same line. Is it OK?

What do you consider "too large"? It looks like it would fit fine on your screenshot. Vertical screenspace is more important than horizontal.

shivamdixit commented:
Sure, I'll change it. (It looks little clumsy in my vertical monitor, for horizontal its perfectly fine)

shivamdixit commented:
Refactored the code. And changed the display text. Screenshot:

![qunit](https://cloud.githubusercontent.com/assets/3061095/6559255/95abf042-c6a7-11e4-9a80-9633a5ecad5d.png)

jzaefferer commented on reporter/html.js:
```diff
@@ -453,10 +453,14 @@ function storeFixture() {
 		userAgent.innerHTML = "";
-		userAgent.appendChild( document.createTextNode( navigator.userAgent ) );
+		userAgent.appendChild(
+			document.createTextNode( navigator.userAgent + ": " + versionInfo )
```
Let's replace the colon with a semicolon. Could also inline the `versionInfo` variable.

jzaefferer commented on src/core.js:
```diff
@@ -184,6 +184,10 @@ config.modules.push( config.currentModule );
 	QUnit.isLocal = location.protocol === "file:";
+
+	// Expose the current QUnit version
+	// `version` initialized at top of scope
```
This comment is now outdated and can be removed

jzaefferer commented on src/qunit.css:
```diff
@@ -11,17 +11,17 @@
 /** Font Family and Sizes */
 
-#qunit-tests, #qunit-header, #qunit-banner, #qunit-testrunner-toolbar, #qunit-userAgent, #qunit-testresult {
+#qunit-tests, #qunit-header, #qunit-banner, #qunit-testrunner-toolbar, #qunit-version, #qunit-userAgent, #qunit-testresult {
```
This looks like it can be reverted, since we're not creating a separate element anymore.

jzaefferer commented on src/qunit.css:
```diff
@@ -68,6 +68,13 @@
 	overflow: hidden;
 }
 
+#qunit-version {
```
Same here, and for the other changes above.

jzaefferer commented:
Thanks, the update is an improvement. Just need to fix a few details.

shivamdixit commented:
I'm not sure why the build is failing. Is the browserstack test are failing because of the UI changes that I made?

jzaefferer commented:
Looks like the Travis build now passed. We run tests on browserstack, which sometimes fails with timeouts. Those are usually non-deterministic, which makes them difficult to debug and fix. Any persistent failures should be obvious enough though...

jzaefferer commented:
This PR now looks good to land to me. Will let @leobalter take a look as well.

leobalter commented:
Looks great, I have only one question: wouldn't it be better to show the QUnit version before the User Agent? It's a smaller text and would help locating every information on that line faster.

```js
"QUnit " + QUnit.version + " @ " + navigator.userAgent

jzaefferer commented:
I had considered the same, since you brought it up: Yes, let's switch the order.

shivamdixit commented:
Changed the order.
![qunit](https://cloud.githubusercontent.com/assets/3061095/6588360/a51d63f2-c7b6-11e4-8f96-f0072ba19286.png)

jzaefferer commented:
I missed this in Leo's comment: Let's stick with the semicolon as the separator. "@" works fine when it refers to the version of something, like "npm install package@version", but here the QUnit version and userAgent are independent. (bikeshedded this with @arthurvr on IRC as well)

shivamdixit commented:
@jzaefferer Okay. Fixed it.

jzaefferer commented:
Thanks! Merged this, squashing commits into one. I also made a tiny edit, removing the space in front of the semicolon.
